documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
-They can be set aside if it means that all hard goals are met.  
-* *Main goals* are inherited from Cruise Control. Some are preset as hard goals.
-Main goals are used in optimization proposals by default.  
-* *Default goals* are the same as the main goals by default. 
-You can specify your own set of default goals.  
-* *User-provided goals* are a subset of default goals that are configured for generating a specific optimization proposal. 
---
-
-Optimization proposals::  Optimization proposals comprise the goals you want to achieve from a rebalance. 
-You generate an optimization proposal to create a summary of proposed changes and the results that are possible with the rebalance.
-The goals are assessed in a specific order of priority.
-You can then choose to approve or reject the proposal.
-You can reject the proposal to run it again with an adjusted set of goals.  
-+
-You can generate an optimization proposal in one of three modes.
-+
---
-* *`full`* is the default mode and runs a full rebalance.
-* *`add-brokers`* is the mode you use after adding brokers when scaling up a Kafka cluster.  
-* *`remove-brokers`* is the mode you use before removing brokers when scaling down a Kafka cluster.
---
-
-Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+Cruise Control comprises four main components: 
+
+Load Monitor:: Load Monitor collects the metrics and analyzes cluster workload data.
+Analyzer:: Analyzer generates optimization proposals based on collected data and configured goals.
+Anomaly Detector:: Anomaly Detector identifies and reports irregularities in cluster behavior.
+Executor:: Executor applies approved optimization proposals to the cluster.
+
+Cruise Control also provides a REST API for client interactions, which Strimzi uses to support these features:
+
+* Generating optimization proposals from optimization goals
+* Rebalancing a Kafka cluster based on an optimization proposal
+* Changing topic replication factor
+
+NOTE: Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+
+== Optimization goals
+
+Optimization goals define objectives for rebalancing, such as distributing topic replicas evenly across brokers. 
+
+They are categorized as follows:
+
+* *Main goals* are a list of goals supported by the Cruise Control instance and used in its operations. 
+By default, this list includes all built-in goals. 
+For a goal to be used in other categories, such as default or hard goals, it must first be listed in main goals. 
+To prevent a goal’s usage, remove it from this list.
+* *Hard goals* are preset and must be satisfied for a proposal to succeed.
+* *Soft goals* are objectives that are prioritized during optimization as much as possible, without preventing a proposal from being created if all hard goals are satisfied.
```

Comment:
```suggestion
* *Soft goals* are objectives that are prioritized during optimization as much as possible, but won't prevent the generation of a proposal if not satisfied. 
```

Reply:
IIRC, at this time, both hard goals and soft goals are preset by upstream Cruise Control. Basically, whatever is not a `hard.goal` (whatever `goals` not listed in the default `hard.goals` config) are soft goals. 

I am not completely sure why Cruise Control offers this `hard.goals` list in the first place if the goal type is hard coded/preset by upstream Cruise Control. Listing a goal in the `hard.goals` list does not change it into a hard goal from what I remember (but my memory is hazy so let's leave this information out for now, I just wanted people to be aware of it)

I bring this up because it may be worth mentioning that soft goals are preset as well